DateCalculation.js

```diff
@@ -0,0 +1,198 @@
+        this.#calculateResultsBtn = document.querySelector('.calculateResults');
+        this.#generalResult = document.querySelector('.dateResult');
+
+        this.#startDate.addEventListener('change', this.#setStartDateCalculation.bind(this));
+        this.#endDate.addEventListener('change', this.#setEndDateCalculation.bind(this));
+        this.#daysRangePreset.addEventListener('change', this.#setDaysRangePreset.bind(this));
+        this.#dayType.addEventListener('change', this.#calculateDayType.bind(this));
+        this.#unitType.addEventListener('change', this.#calculateUnitType.bind(this));
+        this.#calculateResultsBtn.addEventListener('click', () => {
+            this.#calculateDateDifference.bind(this);
+            this.#saveDateInLocalStorage();
+        });
+        this.#dateDifferenceContainer.addEventListener('change', this.#verifyFormCompletion.bind(this));
+
+        this.#verifyFormCompletion();
+        this.#loadDateFromLocalStorage();
+    }
+
+    #setStartDateCalculation() {
+        let startDate = this.#startDate.value;
+        this.#endDate.min = startDate;
+        return startDate;
+    }
+
+    #setEndDateCalculation() {
+        let endDate = this.#endDate.value;
+        this.#startDate.max = endDate;
+        return endDate;
+    }
+
+    #formatDate(date) {
+        const year = date.getFullYear();
+        const month = String(date.getMonth() + 1).padStart(2, '0');
+        const day = String(date.getDate()).padStart(2, '0');
+        return `${year}-${month}-${day}`;
+    }
+
+    #setDaysRangePreset() {
+        let dayRangePreset = parseInt(this.#daysRangePreset.value);
+        if (!isNaN(dayRangePreset)) {
+            let startDate = new Date(this.#startDate.value);
+            if (isNaN(startDate.getTime())) {
+                alert('Будь ласка оберіть початкову дату');
+                this.#daysRangePreset.value = 'default';
+                return;
+            }
+            let endDate = new Date(startDate);
+            endDate.setDate(endDate.getDate() + dayRangePreset);
+            this.#endDate.value = this.#formatDate(endDate);
+        }
+    }
+
+    #calculateWorkingDays(startDate, endDate) {
+        let i = 0;
+        for (let currentDate = new Date(startDate); currentDate <= endDate; currentDate.setDate(currentDate.getDate() + 1)) {
+            let day = currentDate.getDay();
+            if (day !== 0 && day !== 6) {
+                i++;
+            }
+        }
+        return i;
+    }
+
+    #calculateWeekendDays(startDate, endDate) {
+        let i = 0;
+        for (let currentDate = new Date(startDate); currentDate <= endDate; currentDate.setDate(currentDate.getDate() + 1)) {
+            let day = currentDate.getDay();
+            if (day === 0 || day === 6) {
+                i++;
+            }
+        }
+        return i;
+    }
+
+    #calculateDayType() {
+        let startDate = new Date(this.#startDate.value);
```

Comment:
Все, що не перезаписується, через константи
```suggestion
        const startDate = new Date(this.#startDate.value);
```

Reply:
тут і далі по коду в подібних місцях